sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileIndexSuite.scala

```diff
@@ -488,6 +489,26 @@ class FileIndexSuite extends SharedSparkSession {
     val fileIndex = new TestInMemoryFileIndex(spark, path)
     assert(fileIndex.leafFileStatuses.toSeq == statuses)
   }
+
+  test("expire FileStatusCache if TTL is configured") {
+    val sparkConfWithTTl = sparkConf.set(SQLConf.METADATA_CACHE_TTL.key, "1")
+
+    withSparkSession(SparkSession.builder.config(sparkConfWithTTl).getOrCreate()) { spark =>
+      val path = new Path("/tmp", "abc")
```

Comment:
plz use `withTempDir` instead.

Reply:
in this test, we don't need to have files or directories to be created, it's just metadata. So I believe `withTempDir` is not required.

Reply:
Ah, I see. How about `/tmp` -> `/dummy_tmp` make it clear that it is not used.

Reply:
Fixed in https://github.com/apache/spark/pull/28852/commits/18feeb02b11407b9eea1457139a91cb40ddc40cf